flink-mesos/src/main/java/org/apache/flink/mesos/runtime/clusterframework/MesosResourceManager.java

```diff
@@ -312,6 +312,48 @@ private void recoverWorkers() throws Exception {
 	}
 
 	@Override
+	public void shutDown() throws Exception {
+		// shut down all components
+		if (taskMonitor != null) {
+			try {
+				actorSystem.stop(taskMonitor);
+			} catch (Throwable tt) {
+				LOG.error("Failed to stop taskMonitor", tt);
+			}
+		}
+
+		if (connectionMonitor != null) {
+			try {
+				actorSystem.stop(connectionMonitor);
+			} catch (Throwable tt) {
+				LOG.error("Failed to stop connectionMonitor", tt);
+			}
+		}
+
+		if (launchCoordinator != null) {
+			try {
+				actorSystem.stop(launchCoordinator);
+			} catch (Throwable tt) {
+				LOG.error("Failed to stop launchCoordinator", tt);
+			}
+		}
+
+		if (reconciliationCoordinator != null) {
+			try {
+				actorSystem.stop(reconciliationCoordinator);
+			} catch (Throwable tt) {
+				LOG.error("Failed to stop reconciliationCoordinator", tt);
+			}
+		}
+
+		if (actorSystem != null) {
+			actorSystem.shutdown();
+			actorSystem.awaitTermination();
+		}
```

Comment:
We cannot shutdown the `ActorSystem` here because it can be used by someone else outside of the `MesosResourceManager`.